### PR TITLE
Use ``UninferableBase`` instead of ``Uninferable``

### PR DESCRIPTION
**Pull request by DanielNoord**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

Blocked on the new `astroid` release.

### CODE REVIEW

**Review by Pierre-Sassoulas — COMMENTED** [review]
I thought UninferableBase was for typing purpose only ? I like the ``is Uninferable`` a lot, it's pretty pythonic imo.

**Comment by DanielNoord** [comment]
> I thought UninferableBase was for typing purpose only ? I like the `is Uninferable` a lot, it's pretty pythonic imo.

We made a similar change in `astroid` as well, see https://github.com/PyCQA/astroid/pull/1741.

The `isinstance` check helps actually narrow down the type of these variables for `mypy` and other tools so I guess it makes sense to do this here as well.

**Comment by jacobtylerwalls** [comment]
Nice FP fixes in the primer comment related to #5288/#8385.

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 1]
```diff
@@ -1569,7 +1571,7 @@ def _append_context_managers_to_stack(self, node: nodes.Assign) -> None:
         else:
             assignees = [node.targets[0]]
             values = [node.value]
-        if Uninferable in (assignees, values):
+        if any(isinstance(n, UninferableBase) for n in (assignees, values)):
```
Comment (jacobtylerwalls) [inline pylint/checkers/refactoring/refactoring_checker.py:1574]: Unrelated to this PR, but this looks fishy. Why would Uninferable be in a tuple of lists? 🤔

**Review by jacobtylerwalls — DISMISSED** [review]

File: pylint/checkers/refactoring/refactoring_checker.py (hunk 1 above)
Comment (DanielNoord) [reply pylint/checkers/refactoring/refactoring_checker.py:1574]: Yeah.. I guess `elts` is `Uninferable` sometimes? Or this is just 100% broken..

Comment (jacobtylerwalls) [reply pylint/checkers/refactoring/refactoring_checker.py:1574]: Would you mind opening a follow up issue to investigate?

**Review by jacobtylerwalls — DISMISSED** [review]

**Review by Pierre-Sassoulas — DISMISSED** [review]
👍 But blocked by #8387 I guess.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**